pkg/proxy/iptables/proxier_test.go

```diff
@@ -521,194 +521,23 @@ func NewFakeProxier(ipt utiliptables.Interface) *Proxier {
-			// Match both dest IP and dest Port
-			rules: []iptablestest.Rule{
-				{"-d ": "1.2.3.4/32", "--dport ": "80", "-p ": "tcp", "-j ": "KUBE-XLB-GF53O3C2HZEXL2XN"},
-			},
-			destChain: "KUBE-XLB-GF53O3C2HZEXL2XN",
-			destIP:    "1.2.3.4",
-			destPort:  80,
-			expected:  true,
-		},
-		"case 4": {
-			// Match dest IP but doesn't match dest Port
-			rules: []iptablestest.Rule{
-				{"-d ": "1.2.3.4/32", "--dport ": "80", "-p ": "tcp", "-j ": "KUBE-XLB-GF53O3C2HZEXL2XN"},
-			},
-			destChain: "KUBE-XLB-GF53O3C2HZEXL2XN",
-			destIP:    "1.2.3.4",
-			destPort:  8080,
-			expected:  false,
-		},
-		"case 5": {
-			// Match dest Port but doesn't match dest IP
-			rules: []iptablestest.Rule{
-				{"-d ": "1.2.3.4/32", "--dport ": "80", "-p ": "tcp", "-j ": "KUBE-XLB-GF53O3C2HZEXL2XN"},
-			},
-			destChain: "KUBE-XLB-GF53O3C2HZEXL2XN",
-			destIP:    "10.20.30.40",
-			destPort:  80,
-			expected:  false,
-		},
-		"case 6": {
-			// Match the 2nd rule(dest IP)
-			rules: []iptablestest.Rule{
-				{"-d ": "10.20.30.41/32", "-p ": "tcp", "-j ": "REJECT"},
-				{"-d ": "1.2.3.4/32", "-p ": "tcp", "-j ": "REJECT"},
-				{"--dport ": "3001", "-p ": "tcp", "-j ": "REJECT"},
-			},
-			destChain: "REJECT",
-			destIP:    "1.2.3.4",
-			destPort:  8080,
-			expected:  true,
-		},
-		"case 7": {
-			// Match the 2nd rule(dest Port)
-			rules: []iptablestest.Rule{
-				{"-d ": "10.20.30.41/32", "-p ": "tcp", "-j ": "REJECT"},
-				{"--dport ": "3001", "-p ": "tcp", "-j ": "REJECT"},
-			},
-			destChain: "REJECT",
-			destIP:    "1.2.3.4",
-			destPort:  3001,
-			expected:  true,
-		},
-		"case 8": {
-			// Match the 1st rule(dest IP)
-			rules: []iptablestest.Rule{
-				{"-d ": "10.20.30.41/32", "-p ": "tcp", "-j ": "REJECT"},
-				{"--dport ": "3001", "-p ": "tcp", "-j ": "REJECT"},
-			},
-			destChain: "REJECT",
-			destIP:    "10.20.30.41",
-			destPort:  8080,
-			expected:  true,
-		},
-		"case 9": {
-			rules: []iptablestest.Rule{
-				{"-j ": "KUBE-SEP-LWSOSDSHMKPJHHJV"},
-			},
-			destChain: "KUBE-SEP-LWSOSDSHMKPJHHJV",
-			destIP:    "",
-			destPort:  0,
-			expected:  true,
-		},
-		"case 10": {
-			rules: []iptablestest.Rule{
-				{"-j ": "KUBE-SEP-FOO"},
-			},
-			destChain: "KUBE-SEP-BAR",
-			destIP:    "",
-			destPort:  0,
-			expected:  false,
-		},
-	}
-
-	for k, tc := range testCases {
-		if got := hasJump(tc.rules, tc.destChain, tc.destIP, tc.destPort); got != tc.expected {
-			t.Errorf("%v: expected %v, got %v", k, tc.expected, got)
+		if line[0] == '*' {
+			inRightTable = (line == "*"+table)
 		}
-	}
-}
-
-func hasDNAT(rules []iptablestest.Rule, endpoint string) bool {
-	for _, r := range rules {
-		if r[iptablestest.ToDest] == endpoint {
-			return true
+		if inRightTable && line[0] == '-' {
+			rules++
 		}
 	}
-	return false
-}
 
-func errorf(msg string, rules []iptablestest.Rule, t *testing.T) {
-	for _, r := range rules {
-		t.Logf("%q", r)
-	}
-	t.Errorf("%v", msg)
+	// For some reason the metric is off by one
```

Comment:
`countRules` is correct. It's the metric itself that has the wrong value. If you have 10 rules, the value of the metric will be `11.0`.

It looks like this is because we're counting the `COMMIT` line as a rule. We could fix that, but maybe some people are already correcting for this and we'd break them if we fixed it? Or maybe that doesn't matter becuase metrics aren't APIs?

Reply:
I think the metric is buggy.  We should fix the bug

Reply:
it is buggy, I didn't think at that time the precision was important , sorry
https://github.com/kubernetes/kubernetes/pull/99653#discussion_r588448513

Reply:
no, it is that metric